doc/iterations/2023.6/shapes/shape Odesign converter.adoc

```diff
@@ -0,0 +1,46 @@
+= (M) ODesign reader
+
+== Problem
+
+We can create representations descriptions programmatically, with the view model and with the compatibility layer that converts odesign files in representation descriptions.
+It will be more efficient to concentrate our efforts in the creation of representation description programmatically and with the view model.
+As such the odesign will be converted in a view model. We will be able to support more concepts of the odesign model in the view model in a near future (see https://github.com/eclipse-sirius/sirius-components/issues/1920)
+
+Using plain old Java code, we should be able to transform an odesign model into a view model while converting the diagram and form descriptions. 
+
+== Key Result
+
+We will use the Flow model to validate the scenario below (https://github.com/eclipse-sirius/sirius-components/issues/1922).
```

Comment:
While I agree in theory, it looks like you are saying that during this iteration, you have four tasks:

- convert an odesign to a view model
- simplify the programmatic creation of view models
- support additional features in view to match what's available in the odesign compatibility layer
- migrate flow to prove that everything is working and we can get rid of our odesign compatibility layer

Here it feels that none of the steps 1 to 3 can be done without 4.
My gut feeling is that this would end up with all the tasks being started at the same time and I'm not even sure that you could create a relevant commit for each task specifically.

So if some task is required to be done before another, it needs to be explicit first (which comes bask to the fact that you should pretty much create one PR with all the shapes at once).

Now, back to the question, how can we ensure that the commit that you will deliver to provide the converter works?

Reply:
Additional question that needs to be answered:

- Are you planning to only deliver a converter when all the features of the odesign compatibility layer have been migrated to the view DSL?